Client/Frontend/Reader/ReaderMode.js

```diff
@@ -46,29 +46,180 @@ var _firefox_ReaderMode = {
+        template += '    <ul class="dropdown">\n';
+        template += '      <li><a class="dropdown-toggle button style-button" href="#"></a></li>\n';
+        template += '      <li class="dropdown-popup">\n';
+        template += '        <ul id="font-type-buttons"></ul>\n';
+        template += '        <hr></hr>\n';
+        template += '        <ul id="font-size-buttons" class="segmented-button"></ul>\n';
+        template += '        <hr></hr>\n';
+        template += '        <ul id="color-scheme-buttons" class="segmented-button"></ul>\n';
+        template += '      </li>\n';
+        template += '    </ul>\n';
+        template += '    <li><a id="toggle-button" class="button toggle-button" href="#"></a></li>\n';
+        template += '  </ul>\n';
+        template += '\n';
+        template += '</body>\n';
+        template += '\n';
+        template += '</html>\n';
+        return template;
+    },
+    
+    // TODO The following code only makes sense in about:reader context. It may be a good idea to move
+    //   it out of this file and into for example a Reader.js.
+    
+    currentStyle: null,
+    
+    setStyle: function(style) {
+        // Configure the theme (light, dark)
+        if (_firefox_ReaderMode.currentStyle != null) {
+            document.body.classList.remove(_firefox_ReaderMode.currentStyle.theme);
+        }
+        document.body.classList.add(style.theme);
+        
+        // Configure the font size (1-5)
+        if (_firefox_ReaderMode.currentStyle != null) {
+            document.body.classList.remove("font-size" + currentStyle.fontSize);
+        }
+        document.body.classList.add("font-size" + style.fontSize);
+
+        // Configure the font type
+        if (_firefox_ReaderMode.currentStyle != null) {
+            document.body.classList.remove(currentStyle.fontType);
+        }
+        document.body.classList.add(style.fontType);
+        
+        // Remember the style
+        _firefox_ReaderMode.currentStyle = style;
+    },
+
+    _BLOCK_IMAGES_SELECTOR: ".content p > img:only-child, " +
+        ".content p > a:only-child > img:only-child, " +
+        ".content .wp-caption img, " +
+        ".content figure img",
+    
+    updateImageMargins: function() {
+        var contentElement = document.getElementById('reader-content');
+        
+        var windowWidth = window.innerWidth;
+        var contentWidth = contentElement.offsetWidth;
+        var maxWidthStyle = windowWidth + "px !important";
+        
+        var setImageMargins = function(img) {
+            if (!img._originalWidth) {
+                img._originalWidth = img.offsetWidth;
+            }
+            
+            var imgWidth = img._originalWidth;
+            
+            // If the image is taking more than half of the screen, just make
+            // it fill edge-to-edge.
+            if (imgWidth < contentWidth && imgWidth > windowWidth * 0.55) {
+                imgWidth = windowWidth;
```

Comment:
I assume this will break going from landscape -> portrait (or vice-versa). I guess we'll need some orientation change listener?